keps/487-kubectl-plugin/README.md

```diff
@@ -0,0 +1,573 @@
+and make progress.
+-->
+
+## Proposal
+
+We propose creating a command-line tool that can serve as a Kubectl plugin that exposes this missing information. We also propose using a design strategy that mimics existing tools that are available for other workload managers across HPC and cloud that users are comforable with to ease adoption of both the tool and approach of submitting workloads to Kubernetes. The main command-line interactions will take the following shape:
+
+```bash
+# As a Kubectl plugin
+kubectl kueue <options> <command< <args>
+
+# As a standalone command-line tool
+kuctl <options> <command> <args>
+```
+
+For comparison, the Armada team developed a [general client](https://github.com/armadaproject/armada/tree/master/cmd/armadactl) that uses this strategy to be renamed and fold into kubectl. This is an ideal approach because it presents an interface to kueue to make it feel more like a standalone job manager (and not force folks to live in Kubernetes abstractions, at least entirely). 
+
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a user submitting workloads, I want to easily see the status of an entire Workload,
+or check on why my workload is not being admitted. I can do this with the new proposed plugin.
+
+#### Story 2
+
+As a user coming from High Performance Computing, I am not comfortable with using
+`kubectl` and don't want to learn an entirely new means to interact with workloads.
+The proposed plugin makes the transition much easier for me.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+Maintaining the plugin in-sync with the latest version of Kueue will add additional
+maintainer responsibilities.
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
```

Comment:
Throughout the examples, what is missing is a parameter for the apigroup/kind. We probably don't want the client to have to do multiple API calls (corresponding to different kinds) to find one that matches the provided name. Also the name could repeat among different Kinds.

Reply:
Do we want to do:

```
kueue <command> --kind MPIJob
kueue <command> -k MPIJob
```
or

```
kueue <command> --kind kubeflow.org/MPIJob
kueue <command> -k kubeflow.org/MPIJob
```

Or some combination of those (or something else)?

Reply:
I guess it can be similar to how `kubectl get` works: you just say `kubectl get mpijob`. I think this uses the discovery API to learn about the CRDs.

Reply:
> In general, kinds can repeat across different API groups. But for the purpose of simplifying users' life, we could have some aliases for "well-known" APIs.

How often does that happen? Let's say we have a kubeflow.org/MPIJob and dinosaur.org/MPIJob and I ask for MPIJob. I would get both. I would then likely see the API group, and I could further adjust my filter to include API group as well.